File: src/main/java/play/pluv/playlist/infra/OcrApiClient.java
package play.pluv.playlist.infra;

import java.util.List;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.service.annotation.PostExchange;
import play.pluv.playlist.infra.dto.OcrMusicRequest;
import play.pluv.playlist.infra.dto.OcrMusicResponse;

public interface OcrApiClient {

  @PostExchange("http://192.168.0.26:8000/ocr")
  List<OcrMusicResponse> ocrPlayListImage(@RequestBody final OcrMusicRequest request);
}

File: src/main/kotlin/play/pluv/playlist/infra/OcrApiClient.kt
package play.pluv.playlist.infra

import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.service.annotation.PostExchange
import play.pluv.playlist.infra.dto.OcrMusicRequest
import play.pluv.playlist.infra.dto.OcrMusicResponse

interface OcrApiClient {

    @PostExchange("http://192.168.0.26:8000/ocr")
    fun ocrPlayListImage(@RequestBody request: OcrMusicRequest): List<OcrMusicResponse>
}
